eos-repartition-mbr

```diff
@@ -0,0 +1,160 @@
+  parts=$(echo "$parts" | sed '$d')
+elif [ $lastpart -gt $swap_partno ]; then
+  echo "repartition: found $lastpart partitions?"
+  exit 1
+fi
+
+find_and_remove_partition() {
+	local name=${1:?No name supplied to ${FUNCNAME}}
+	local guid=${2:?No GUID supplied to ${FUNCNAME}}
+
+	local partno=$(echo "$parts" | sed -n -e "s/[^:]*\([0-9]\) :.*$guid.*/\1/p")
+
+	if [ -z "${partno}" ]; then
+		echo "$0: $name not found"
+		exit 1
+	fi
+
+	local part=${root_disk}${partno}
+	local part_dev=${part#/dev/}
+
+	parts=$(echo "$parts" | sed -e "/$part_dev/d")
+}
+
+find_and_remove_partition "ESP" "C12A7328-F81F-11D2-BA4B-00A0C93EC93B"
+find_and_remove_partition "BIOS boot" "21686148-6449-6E6F-744E-656564454649"
+
+# calculate new partition sizes
+disk_size=$(blockdev --getsz $root_disk)
+part_size=$(blockdev --getsz $root_part)
+part_start=$(echo "$parts" | sed -n -e '$ s/.*start=[ ]\+\([0-9]\+\).*$/\1/p')
+part_end=$(( part_start + part_size ))
+echo "Dsize $disk_size Psize $part_size Pstart $part_start Pend $part_end"
+
+# Calculate the new root partition size, assuming that it will expand to fill
+# the remainder of the disk
+new_size=$(( disk_size - part_start ))
+
+# If we find ourselves with >100GB free space, we'll use the final 4GB as
+# a swap partition
+added_space=$(( new_size - part_size ))
+if [ $added_space -gt 209715200 ]; then
+  new_size=$(( new_size - 8388608 ))
+  swap_start=$(( part_start + new_size ))
+
+  # Align swap partition start to 1MB boundary
+  residue=$(( swap_start % 2048 ))
+  if [ $residue -gt 0 ]; then
+    swap_start=$(( swap_start + 2048 - residue ))
+    # might as well bump up root partition size too, instead of leaving a gap
+    new_size=$(( new_size + 2048 - residue ))
+  fi
+fi
+
+if [ $new_size -gt $part_size ]; then
+  echo "Try to resize $root_part to fill $new_size sectors"
+  parts=$(echo "$parts" | sed -e "$ s/size=[0-9\t ]*,/size=$new_size,/")
+fi
+
+if [ -n "$swap_start" ]; then
+  # Create swap partition
+  echo "Create swap partition at $swap_start"
+  parts="$parts
+start=$swap_start, type=82"
+fi
+
+# Reset partition indexes
+root_disk_dev=${root_disk#/dev/}
+parts=$(echo "$parts" | sed -e "s/${root_disk_dev}.//")
+
+# GPT -> DOS
+parts=$(echo "$parts" | sed -e "s/label: gpt/label: dos/")
+
+# Set MBR partition type
+parts=$(echo "$parts" | sed -e "s/, type=$dps_root_guid/, type=83/")
+# Remove partition UUIDs
+parts=$(echo "$parts" | sed -e "s/, uuid=\S\+//g")
+
+echo "$parts"
+echo "$parts" | sfdisk --force --no-reread $root_disk
+udevadm settle
+
+# mkswap with the new partition index
+swap_partno=$((swap_partno - 2))
+swap_part=${root_disk}${swap_partno}
+[ -e "$swap_part" ] && mkswap -L eos-swap $swap_part
+
+# Remove GPT primary partition table
+dd if=/dev/zero of=${root_disk} conv=fsync bs=512 count=1 seek=1
+
+# Remove GPT backup partition table
+dd if=/dev/zero of=${root_disk} conv=fsync bs=512 count=1 seek=$((part_size - 1))
```

Comment:
I'm very suspicious about this line. I think the backup table is 1 sector from the end of the *disk*, not the partition; so I think this is actually zeroing 512 bytes of the root filesystem.